### PR TITLE
send NODE command tests

### PR DESCRIPTION
**Pull request by spivachuk**
- Added a test verifying that "send NODE" command fails if hex key is used instead of base58 key.
- Added a test verifying that "send NODE" command succeeds when "SERVICES" parameter has been omitted.
- Extracted some common code from CLI tests to conftest and helper modules.

### CODE REVIEW

File: sovrin_client/test/cli/test_send_node_services_omitted.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    tconf.UpdateGenesisPoolTxnFile = True
+
+    def reset():
+        tconf.UpdateGenesisPoolTxnFile = oldVal
+
+    request.addfinalizer(reset)
+    return tconf
+
+
+def testNewNodeAddedWhenServicesParamOmitted(be, do, poolNodesStarted,
```
Comment (lovesh) [inline sovrin_client/test/cli/test_send_node_services_omitted.py]: The test `testSuspendNodeWhichWasNeverActive` already checks this, why a new test?

File: sovrin_client/test/cli/test_send_node_services_omitted.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    del newNodeVals['newNodeData'][SERVICES]
+
+    do('send NODE dest={newNodeIdr} data={newNodeData}',
+       within=8, expect=['Node request completed'], mapper=newNodeVals)
+
+    # Verify that the new node has been actually added to the pool
+    newNodeData = newNodeVals['newNodeData']
+
+    def checkClientConnected(client):
```
Comment (lovesh) [inline sovrin_client/test/cli/test_send_node_services_omitted.py]: Being in `nodeReg` does not mean its connected. We have methods to check if client is connected to all nodes or not. Also a client should not attempt to connect to a node without `services` entry so they should not end up in `nodeReg`. We have a jira ticket for that.

File: sovrin_client/test/cli/test_send_node_services_omitted.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        name = newNodeData[ALIAS] + CLIENT_STACK_SUFFIX
+        assert name in client.nodeReg
+
+    def checkNodeConnected(nodes):
```
Comment (lovesh) [inline sovrin_client/test/cli/test_send_node_services_omitted.py]: Being in `nodeReg` does not mean its connected. We have methods to check if nodes are inter-connected. Also a node should not attempt to connect to a node without services entry

File: sovrin_client/test/cli/test_send_node_services_omitted.py (hunk 2 above)
Comment (spivachuk) [inline sovrin_client/test/cli/test_send_node_services_omitted.py]: I took this verification from the existing fixture `newNodeAdded` (it was just moved to `conftest.py` in scope of this PR).

File: sovrin_client/test/cli/test_send_node_services_omitted.py (hunk 3 above)
Comment (spivachuk) [inline sovrin_client/test/cli/test_send_node_services_omitted.py]: Also took this verification from the existing fixture `newNodeAdded`.

File: sovrin_client/test/cli/test_send_node_services_omitted.py (hunk 1 above)
Comment (spivachuk) [inline sovrin_client/test/cli/test_send_node_services_omitted.py]: Was not aware that `testSuspendNodeWhichWasNeverActive` verifies the same case. So I'll remove `testNewNodeAddedWhenServicesParamOmitted`.

File: sovrin_client/test/cli/test_send_node_services_omitted.py (hunk 2 above)
Comment (spivachuk) [inline sovrin_client/test/cli/test_send_node_services_omitted.py]: > Also a client should not attempt to connect to a node without services entry so they should not end up in nodeReg.
Got it. I'll remove this test anyway (as discussed in the thread above).

File: sovrin_client/test/cli/test_send_node_hex_instead_of_base58.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+from plenum.common.util import cryptonymToHex
+from sovrin_client.test.cli.helper import getNewNodeVals
+from sovrin_common.roles import Roles
+
+
+@pytest.fixture(scope='module')
+def tconf(tconf, request):
+    oldVal = tconf.UpdateGenesisPoolTxnFile
+    tconf.UpdateGenesisPoolTxnFile = True
```
Comment (lovesh) [inline sovrin_client/test/cli/test_send_node_hex_instead_of_base58.py]: Why are you setting this? And `UpdateGenesisPoolTxnFile` is deprecated anyway

**Comment by lovesh** [comment]
@spivachuk You can unskip that test as the corresponding code has been merged in plenum master

**Comment by spivachuk** [comment]
@lovesh I'll unskip it when having plenum-dev 0.3.70 used in sovrin-client-dev.

**Pull request merged**